### PR TITLE
Added logging utilities for projects

### PR DESCRIPTION
**Pull request by SimonDarksideJ**
# Reality Collective - Reality Toolkit Pull Request

## Overview

Added some new logging functionality to help provide a more consistent logging system, which can also be captured for exporting.

### CODE REVIEW

**Comment by FejZa** [comment]
I don't understand what this adds to the built-in log. Requiring a MonoBehaviour to capture log seems a bit odd and error prone.

**Review by FejZa — CHANGES_REQUESTED** [review]

File: Runtime/Utilities/CaptureApplicationLog.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+using UnityEngine;
+
+namespace RealityCollective.Utilities
+{
+    /// <summary>
+    /// Helper function to attach to a persistent gameobject in the loading scece, relays Unity events to the logging entity
+    /// </summary>
+    public class CaptureApplicationLog : MonoBehaviour
+    {
+        void Awake()
```
Comment (FejZa) [inline Runtime/Utilities/CaptureApplicationLog.cs:14]: Missing private modifier

**Comment by SimonDarksideJ** [comment]
> I don't understand what this adds to the built-in log. Requiring a MonoBehaviour to capture log seems a bit odd and error prone.

You do not "have" to use the Additional Capture MonoBehaviour, it is optional.

What this effectively gives you is a way to capture logging throughout an application into a central log (if you need it), if you also want to capture Unity's own logs, you can use the additional MonoBehaviour.

You can then ship or inspect the captured log.  We used it extensively in my last position to submit logs to an external service for review.

File: Runtime/Utilities/StaticLogger.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        /// <param name="AppLog">Is the log event already in the Unity debugger, if so, skip recording it and just add it to the text log.</param>
+        /// <param name="includeStackTrace">Should a stack trace also be generated when logging.</param>
+        public static void Log(string message, LogType logType = LogType.Log, bool AppLog = false, bool includeStackTrace = true)
+        {
+            if (!pauseLog)
+            {
+                if (!AppLog)
+                {
+                    Console.WriteLine(message);
+                    if (DebugMode)
```
[the remark this answers is not in the document]
Comment (SimonDarksideJ) [reply Runtime/Utilities/StaticLogger.cs:75]: Note the "AppLog" flag, which is captured externally, this prevents repeated logs from being "logged".  Used this extensively without issue previously.

File: Runtime/Utilities/CaptureApplicationLog.cs (hunk 1 above)
Comment (SimonDarksideJ) [reply Runtime/Utilities/CaptureApplicationLog.cs:14]: Fair enough

File: Runtime/Utilities/StaticLogger.cs (hunk 2 above)
[the remark this answers is not in the document]
Comment (FejZa) [reply Runtime/Utilities/StaticLogger.cs:75]: I see. But speaking of that parameter, it should be pascalCase

**Review by FejZa — APPROVED** [review]

**Pull request merged**